overlay.d/21dhcp-chrony/etc/NetworkManager/dispatcher.d/20-coreos-chrony-dhcp

```diff
@@ -20,9 +20,13 @@ export LC_ALL=C
     exit 0
 
 # If a dhclient installation is present, avoid a redundant operation
-# with dhclient which handles NTP server config through its own
-# NetworkManager dispatcher script 11-dhclient.
-[ -e /usr/sbin/dhclient ] && exit 0
+# if dhclient handles NTP server config through its own chrony config script,
+# exec'ed by NetworkManager's 11-dhclient dispatcher script.
+# In RHCOS, 11-dhclient is excluded and not shipped with base image.
+[ -e /usr/sbin/dhclient ] && \
+    [ -e /etc/dhcp/dhclient.d/chrony.sh ] && \
+    [ -e /etc/NetworkManager/dispatcher.d/11-dhclient ] && \
+    exit 0
```

Comment:
@sohankunkerkar any feedback on this?

Reply:
This makes total sense to me.  We could also just test for the dispatcher script I think?

Reply:
I don't mind the robustness the additional checks give us, but I'm not opposed to changing it either.
Either way, I'll update https://src.fedoraproject.org/rpms/chrony/blob/master/f/chrony-nm-dispatcher-dhcp.patch with these changes.

Reply:
It seems reasonable to me. I've not tested these changes though

> Either way, I'll update https://src.fedoraproject.org/rpms/chrony/blob/master/f/chrony-nm-dispatcher-dhcp.patch with these changes.

Yup, that's the right way forward for this kind of change.